File: sdk/nodejs/src/model/system/permission.ts


import { User } from "./user";

import { Role } from "./role";


export const PermissionResource = {
    resource: "Permission",
    namespace: "system",
};

// Sub Types

// Resource Type
export interface Permission {
    id: string;
version: number;
createdBy?: string;
updatedBy?: string;
createdOn?: string;
updatedOn?: string;
namespace?: string;
resource?: string;
property?: string;
propertyValue?: string;
propertyMode?: 'PROPERTY_MATCH_ONLY' | 'PROPERTY_MATCH_ANY';
operation: 'READ' | 'CREATE' | 'UPDATE' | 'DELETE' | 'FULL';
recordIds?: string[];
before?: string;
after?: string;
user?: User;
role?: Role;
permit: 'ALLOW' | 'REJECT';
localFlags?: object;

}
// Resource and Property Names
export const PermissionName = "Permission";

export const PermissionIdName = "Id";

export const PermissionVersionName = "Version";

export const PermissionCreatedByName = "CreatedBy";

export const PermissionUpdatedByName = "UpdatedBy";

export const PermissionCreatedOnName = "CreatedOn";

export const PermissionUpdatedOnName = "UpdatedOn";

export const PermissionNamespaceName = "Namespace";

export const PermissionResourceName = "Resource";

export const PermissionPropertyName = "Property";

export const PermissionPropertyValueName = "PropertyValue";

export const PermissionPropertyModeName = "PropertyMode";

export const PermissionOperationName = "Operation";

export const PermissionRecordIdsName = "RecordIds";

export const PermissionBeforeName = "Before";

export const PermissionAfterName = "After";

export const PermissionUserName = "User";

export const PermissionRoleName = "Role";

export const PermissionPermitName = "Permit";

export const PermissionLocalFlagsName = "LocalFlags";

File: sdk/python/apibrew/model/permission.py
from datetime import datetime
from typing import List, Optional
from enum import Enum
from apibrew.entity import Entity, EntityInfo
from __future__ import annotations

from apibrew.model.user import User
from apibrew.model.role import Role



class AuditData:
    createdBy: str
    updatedBy: str
    createdOn: datetime
    updatedOn: datetime

class BooleanExpression:
    and_: list[BooleanExpression]
    or_: list[BooleanExpression]
    not_: BooleanExpression
    equal: PairExpression
    lessThan: PairExpression
    greaterThan: PairExpression
    lessThanOrEqual: PairExpression
    greaterThanOrEqual: PairExpression
    in_: PairExpression
    isNull: Expression
    regexMatch: RegexMatchExpression

class PairExpression:
    left: Expression
    right: Expression

class RegexMatchExpression:
    pattern: str
    expression: Expression

class Expression:
    property: str
    value: dict


class Operation(Enum):
    READ = "READ"
    CREATE = "CREATE"
    UPDATE = "UPDATE"
    DELETE = "DELETE"
    FULL = "FULL"


class Permit(Enum):
    ALLOW = "ALLOW"
    REJECT = "REJECT"



class Permission:
    id: str
    version: int
    auditData: AuditData
    namespace: str
    resource: str
    recordSelector: BooleanExpression
    operation: Operation
    before: datetime
    after: datetime
    user: User
    role: Role
    permit: Permit
    localFlags: dict

    @staticmethod
    def entity_info():
        return EntityInfo("system", "Permission", "system-permission")




